From 37e8a6a7e5875e20a8de07fbfbb69912f1964f7d Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Mon, 28 Oct 2019 09:47:49 +0100
Subject: [PATCH] Makefile: allow the user to disable stack protector

Allow the user to disable stack-protector by overriding CFLAGS as it is
not supported by all toolchains.

Fixes:
 - http://autobuild.buildroot.net/results/cba1ae830c7a4d1740098fe67aec59b4dc2f9a03

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status: https://github.com/smuellerDD/jitterentropy-library/pull/12]
---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 41bc4f7..b0f20e2 100644
--- a/Makefile
+++ b/Makefile
@@ -1,9 +1,9 @@
 # Compile Noise Source as user space application
 
 CC ?= gcc
-CFLAGS +=-Wextra -Wall -pedantic -fPIC -O0
 #Hardening
-CFLAGS +=-fstack-protector-all -fwrapv --param ssp-buffer-size=4
+CFLAGS ?=-fstack-protector-all --param ssp-buffer-size=4
+CFLAGS +=-Wextra -Wall -pedantic -fPIC -O0 -fwrapv
 LDFLAGS +=-Wl,-z,relro,-z,now
 
 # Change as necessary
-- 
2.23.0

